app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadListFragment.kt

```diff
@@ -76,8 +106,22 @@ class ThreadListFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {
         threadListViewModel.refreshThreads()
     }
 
+    private fun setupMenuDrawer() {
+        binding.drawerLayout.addDrawerListener(drawerListener)
+
+        val fragment = MenuDrawerFragment(
+            closeDrawer = { closeDrawer() },
+            isDrawerOpen = { binding.drawerLayout.isOpen },
+        ).also { menuDrawerFragment = it }
+
+        activity?.supportFragmentManager
+            ?.beginTransaction()
+            ?.add(R.id.menuDrawerFragment, fragment)
+            ?.commit()
+    }
```

Comment:
You have to initialize the `navigationDrawer` only once in the activity and use the `setDrawerLockMode` method to either display it here, and hide it when needed